### PR TITLE
Building source

### PR DESCRIPTION
**Pull request by mbruzek**
This change builds kubernetes from source and hosts the binaries on the nginx web server for the minions to download.

It uses git to clone the source.
The kubernetes makefile uses docker to build the files.
Adding a template file to host the files on nginx.

Please note this works for v0.8.1 but there is a compilation problem with the most recent versions of kubernetes that prevents make all from succeeding in this setup.  I opened an issue about that problem: https://github.com/GoogleCloudPlatform/kubernetes/issues/5098  and will follow up when it gets resolved.

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -2,4 +2,8 @@ options:
   version:
     type: string
     default: "v0.8.1"
-    description: "The kubernetes version to install in the charm."
+    description: |
+        The kubernetes release to use in this charm. The binary files are
+        compiled from the source identified by this tag in gihub. Using the
```
Comment (lazypower) [inline config.yaml]: Typo on github

File: hooks/hooks.py [hunk 2]
```diff
@@ -68,6 +115,13 @@ def relation_changed():
         if render_file(n, template_data) or not host.service_running(n):
             host.service_restart(n)
 
+    # Render the file that makes the kubernetes binaries available to minions.
+    if render_file(
+            'distribution', template_data,
+            'conf.tmpl', '/etc/nginx/sites-enabled/distribution') or \
```
Comment (lazypower) [inline hooks/hooks.py:116]: Will we be adding multiple versions for this? I thought we were going to a vhost per version deployed for distribution. Perhaps i'm remembering this incorrectly.

**Comment by lazypower** [comment]
@mbruzek I left some inline comments, this looks good as a first pass. We can refactor quite a bit of the subprocess shell outs for git with a python gitlib, however this is functional. 

Take a look at the comments and let me know what you think, if there are no further action items, :+1:  LGTM

File: config.yaml (hunk 1 above)
Comment (mbruzek) [inline config.yaml]: Good catch, sorry about the typo, will fix it.

File: hooks/install [hunk 3]
```diff
@@ -1,24 +1,55 @@
...
+    repository = 'https://github.com/GoogleCloudPlatform/kubernetes.git'
+    kubernetes_directory = '/opt/kubernetes'
+
+    command = ['git', 'clone', repository, kubernetes_directory]
+    print(command)
+    output = subprocess.check_output(command)
+    print(output)
+
+    hookenv.log('Install complete')
+
```
Comment (lazypower) [inline hooks/install:55]: We're missing an open-port 8080

**Comment by lazypower** [comment]
:+1:

**Pull request merged**